src/software/ai/hl/stp/tactic/shadow_freekicker_tactic_test.cpp

```diff
@@ -34,6 +35,676 @@ TEST(ShadowFreekickerTacticTest, test_shadow_free_kicker_left_side)
 
     ASSERT_TRUE(action_ptr);
 
+    auto move_action = std::dynamic_pointer_cast<MoveAction>(action_ptr);
+    ASSERT_NE(nullptr, move_action);
+    // The edge of the robot should be slightly more than 0.5m from the ball
+    EXPECT_NEAR((world.ball().position() - move_action->getDestination()).length(), 0.62,
+                0.05);
+
+    // The robot should be just to the right of the line between the enemy robot and
+    // the ball (from the POV of the enemy robot)
+    Line enemy_to_ball_line =
+        Line(world.enemyTeam().getAllRobots().at(0).position(), world.ball().position());
+    EXPECT_NEAR(distance(enemy_to_ball_line, move_action->getDestination()), 0.09, 0.01);
+    Angle enemy_to_ball_angle =
+        (world.ball().position() - world.enemyTeam().getAllRobots().at(0).position())
+            .orientation();
+    Angle enemy_to_dest_angle = (move_action->getDestination() -
+                                 world.enemyTeam().getAllRobots().at(0).position())
+                                    .orientation();
+    EXPECT_LT(enemy_to_dest_angle, enemy_to_ball_angle);
+}
+
+TEST(ShadowFreekickerTacticTest,
+     test_shadow_free_kicker_right_side_with_enemy_robot_at_middle_left_on_field)
+{
+    World world = ::Test::TestUtil::createBlankTestingWorld();
+    world       = ::Test::TestUtil::setEnemyRobotPositions(
+        world, {Point(0, world.field().friendlyCornerPos().y())},
+        Timestamp::fromSeconds(0));
+    world = ::Test::TestUtil::setBallPosition(
+        world, Point(-0.09, world.field().friendlyCornerPos().y() - 0.09),
+        Timestamp::fromSeconds(0));
+    world =
+        ::Test::TestUtil::setBallVelocity(world, Vector(0, 0), Timestamp::fromSeconds(0));
+
+    Robot friendly_robot(0, Point(0, 0), Vector(0, 0), Angle::zero(),
+                         AngularVelocity::zero(), Timestamp::fromSeconds(0));
+    world.mutableFriendlyTeam().updateRobots({friendly_robot});
+
+    ShadowFreekickerTactic tactic =
+        ShadowFreekickerTactic(ShadowFreekickerTactic::RIGHT, world.enemyTeam(),
+                               world.ball(), world.field(), false);
+    tactic.updateRobot(friendly_robot);
+
+    auto action_ptr = tactic.getNextAction();
+
+    ASSERT_TRUE(action_ptr);
+
+    auto move_action = std::dynamic_pointer_cast<MoveAction>(action_ptr);
+    ASSERT_NE(nullptr, move_action);
+    // The edge of the robot should be slightly more than 0.5m from the ball
+    EXPECT_NEAR((world.ball().position() - move_action->getDestination()).length(), 0.62,
+                0.05);
+
+    // The robot should be just to the left of the line between the enemy robot and
+    // the ball (from the POV of the enemy robot)
+    Line enemy_to_ball_line =
+        Line(world.enemyTeam().getAllRobots().at(0).position(), world.ball().position());
+    EXPECT_NEAR(distance(enemy_to_ball_line, move_action->getDestination()), 0.09, 0.01);
```

Comment:
In the same line as the 0.62 comment, it might be beneficial to use constants/defines to denote what 0.09, 0.01, 0.62, etc. mean (as opposed to having magic numbers).

Reply:
Those constants are defined and commented in the test fixture